block/manager.go

```diff
@@ -308,23 +308,49 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context) {
 
 	ticker := time.NewTicker(m.conf.BlockTime)
 
+	var tickerEmptyBlocksMaxTime *time.Ticker
+	var tickerEmptyBlocksMaxTimeCh <-chan time.Time
+	if m.conf.EmptyBlocksMaxTime > 0 {
+		tickerEmptyBlocksMaxTime = time.NewTicker(m.conf.EmptyBlocksMaxTime)
+		tickerEmptyBlocksMaxTimeCh = tickerEmptyBlocksMaxTime.C
+	}
+
+	//Allow the initial block to be empty
+	produceEmptyBlock := true
 	for {
 		select {
+		//Context canceled
 		case <-ctx.Done():
 			return
+		//Empty blocks timeout
+		case <-tickerEmptyBlocksMaxTimeCh:
+			m.logger.Error("No transactions for too long, allowing to produce empty block")
```

Comment:
not sure I understand why would I want to produce an empty block if there were not transactions for too long (assuming the hub can receive empty batches for liveness)?

Reply:
IMO It makes more sense to separate `block production` and `batch submission`
Block production is important for rollapp liveness
while the batch submission is more related to finality

I will add an `expiry timeout` for the batch submission trigger as well

So in a sense,
block_time < EmptyBlocksMaxTime << BatchSubmissionMaxTime
(e.g 200ms < 1m << 1h)

Reply:
It will be handled in this issue
https://github.com/dymensionxyz/dymint/issues/324

Reply:
good thinking with the `batchSubmissionMaxTime`. 

though I think it's already necessary.  if it doesn't exist now (and only batch size does), so it will take the sequencer (assuming empty blocks produced every 1 minute) hours to update the hub and may hurt the uptime as indicated by the hub/portal.  

btw we may do `batchSubmissionMaxBlocks` which measures time by hub blocks.  

I think it makes more sense as this is how the rollapp update rate for slashing etc is going to be checked by the hub.